runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -1256,6 +1256,12 @@ InterpreterEmulator::visitInvokedynamic()
          || comp()->compileRelocatableCode()
       ) return; // do nothing if unresolved, is AOT compilation
    uintptr_t * invokeCacheArray = (uintptr_t *) owningMethod->callSiteTableEntryAddress(callSiteIndex);
+   // CallSite table entry is expected to be an array object upon successful resolution, but this is not
+   // the case when an exception occurs during the invokedynamic resolution, in which case an exception
+   // object is placed in the slot instead
+   if (!comp()->fej9()->isArrayObject(*invokeCacheArray))
```

Comment:
Here too will need to have VM access while looking inside the object `*invokeCacheArray`. 
We acquire VM access later on in 
`fej9->getKnotIndexOfInvokeCacheArrayAppendixElement(comp(), invokeCacheArray);`
called from
`updateKnotAndCreateCallSiteUsingInvokeCacheArray`
Hence, maybe we can tuck the array check in `getKnotIndexOfInvokeCacheArrayAppendixElement`

Reply:
Similar explanation as described in [my other response](https://github.com/eclipse-openj9/openj9/pull/17625#discussion_r1239254436).